engine/src/main/java/org/terasology/rendering/dag/nodes/LightGeometryNode.java

```diff
@@ -99,43 +98,16 @@ public void process() {
             // TODO: find a more elegant way
             worldRenderer.renderLightComponent(lightComponent, worldPosition, lightGeometryShader, false);
         }
-        postRenderCleanupLightGeometry();
-        PerformanceMonitor.endActivity();
-
-    }
-
-    // TODO: figure how lighting works and what this does
-    private void postRenderCleanupLightGeometry() {
-        glDisable(GL_STENCIL_TEST);
-        glCullFace(GL_BACK);
 
-        bindDisplay();
+        PerformanceMonitor.endActivity();
     }
 
     // TODO: figure how lighting works and what this does
     private void preRenderSetupLightGeometry() {
```

Comment:
Let's place the much more simplified content of this method into process(). And of course ideally the glStencil|BlendFunc calls would become StateChanges. If you'd rather do it after the GSOC deadline you can just add a TODO.